src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.cs

```diff
@@ -237,6 +240,45 @@ protected internal static void ValidateMatchTimeout(TimeSpan matchTimeout)
         }
 
         /// <summary>
+        /// Specifies the default RegEx matching timeout value (i.e. the timeout that will be used if no
+        /// explicit timeout is specified).
+        /// The default is queried from the current <code>AppDomain</code>.
+        /// If the AddDomain's data value for that key is not a <code>TimeSpan</code> value or if it is outside the
+        /// valid range, an exception is thrown.
+        /// If the AddDomain's data value for that key is <code>null</code>, a fallback value is returned.
+        /// </summary>
+        /// <returns>The default RegEx matching timeout for this AppDomain</returns>
+        private static TimeSpan InitDefaultMatchTimeout()
+        {
+            // Query AppDomain
+            AppDomain ad = AppDomain.CurrentDomain;
+            object defaultMatchTimeoutObj = ad.GetData(DefaultMatchTimeout_ConfigKeyName);
+
+            // If no default is specified, use fallback
+            if (defaultMatchTimeoutObj == null)
+            {
+                return InfiniteMatchTimeout;
+            }
```

Comment:
There is no test for this null case.

Reply:
Yes because I can't test without adding the file as a link in the test project (which I wanted to avoid). The `DefaultMatchTimeout` static field isn't accessible from the outside and I prefer not to use reflection (because of uapaot issues with it).

Reply:
can't you not set the appdomain regex key value = null?

Reply:
I can of course but I don't need to because the default value is null when it doesn't find anything. The problem is that I can't assert that the `InfiniteMatchTimeout` is then set on the `DefaultMatchTimeOut` field. I can't access that field.